src/Cake.Common.Tests/Unit/Tools/MSBuild/MSBuildSettingsExtensionsTests.cs

```diff
@@ -261,5 +261,41 @@ public void Should_Return_The_Same_Configuration()
                 Assert.Equal(settings, result);
             }
         }
+
+        public sealed class TheWithLoggersMethod
+        {
+            [Fact]
+            public void Should_Add_Logger()
+            {
+                // Given
+                var settings = new MSBuildSettings();
+
+                // When
+                settings.WithLogger("LoggerAssembly1", "LoggerClass1", "LoggerParameters1");
+                settings.WithLogger("LoggerAssembly2", "LoggerClass2", "LoggerParameters2");
+
+                // Then
+                Assert.Equal(2, settings.Loggers.Count);
+                Assert.Equal("LoggerAssembly1", settings.Loggers[0].Assembly);
```

Comment:
ICollection doesn't support indexing, LINQ could be one option istead
I.e.
```csharp
var first = settings.Loggers.First();
var second = settings.Loggers.Skip(1).First();
```
and use those variables for the asserts.